lxd/device/cdi/configure.go

```diff
@@ -0,0 +1,292 @@
+			mount.ContainerPath = strings.TrimPrefix(evaluatedHostPath, rootPath)
+		}
+
+		configDevices.DiskDevs[deviceName] = map[string]string{
+			"type":              "disk",
+			"source":            evaluatedHostPath,
+			"path":              mount.ContainerPath,
+			"shift":             shift,
+			"raw.mount.options": chosenOptsStr,
+		}
+	}
+
+	// If the user desires to run a nested docker container inside a LXD container,
+	// the Tegra CSV files also need to be mounted so that the nvidia docker runtime
+	// can be auto-enabled as 'csv' mode.
+	if cdiID.Vendor == Nvidia && cdiID.Class == IGPU {
+		tegraCSVFilesCandidates := defaultNvidiaTegraCSVFiles(rootPath)
+		tegraCSVFiles := make([]string, 0)
+		for _, candidate := range tegraCSVFilesCandidates {
+			_, err := os.Stat(candidate)
+			if err == nil {
+				tegraCSVFiles = append(tegraCSVFiles, candidate)
+			} else if os.IsNotExist(err) {
+				continue
+			} else {
+				return nil, err
+			}
+		}
+
+		if len(tegraCSVFiles) == 0 {
+			return nil, fmt.Errorf("No CSV files detected for Tegra iGPU")
+		}
+
+		for _, tegraFile := range tegraCSVFiles {
+			splittedDevName = strings.Split(tegraFile, "/")
+			deviceName = splittedDevName[len(splittedDevName)-1]
+			configDevices.DiskDevs[deviceName] = map[string]string{
+				"type":     "disk",
+				"source":   tegraFile,
+				"path":     strings.TrimPrefix(tegraFile, rootPath),
+				"shift":    shift,
+				"readonly": "true",
+			}
+		}
+	}
+
+	return indirectSymlinks, nil
+}
+
+// specHookToLXDCDIHook will translate a hook from a CDI spec into an entry in a `Hooks`.
+// Some CDI hooks are not relevant for LXD and will be ignored.
+func specHookToLXDCDIHook(hook *specs.Hook, hooks *Hooks) error {
+	if hook == nil {
+		return fmt.Errorf("CDI hook is nil")
+	}
+
+	rootPath := ""
+	if shared.InSnap() {
+		rootPath = "/var/lib/snapd/hostfs"
+	}
+
+	switch hook.Args[2] {
+	case "create-symlinks":
+		if len(hook.Args) < 5 {
+			return fmt.Errorf("CDI 'create-symlinks' hook has not enough arguments: %v", hook.Args)
+		}
+
+		for i := 4; i < len(hook.Args); i += 2 {
+			entry := strings.Split(hook.Args[i], "::")
+			if len(entry) != 2 {
+				return fmt.Errorf("Invalid symlink entry %q", hook.Args[i])
+			}
+
+			// `Link` is always an absolute path and `Target` (a `Link` points to a `Target`) is relative
+			// to the `Link` location in the CDI spec. A resolving operation will be needed to have the absolute
+			// path of the `Target`
+			hooks.Symlinks = append(hooks.Symlinks, SymlinkEntry{Target: strings.TrimPrefix(entry[0], rootPath), Link: strings.TrimPrefix(entry[1], rootPath)})
+		}
+
+	case "update-ldcache":
+		// The `update-ldcache` hook can be used without any arguments, but in our case we don't need it.
+		if len(hook.Args) < 5 {
+			logger.Warnf("CDI 'update-ldcache' hook has not enough arguments: %v", hook.Args)
+			return nil
+		}
+
+		for i := 4; i < len(hook.Args); i += 2 {
```

Comment:
Similar comment as above where we should check for the `--folder` argument.

Reply:
fixed